HIP/hip-1056.md

```diff
@@ -0,0 +1,1727 @@
+
+    /**
+     * A state for the `159` upgrade file data
+     */
+    STATE_ID_UPGRADE_DATA_159 = 10010;
+}
+```
+
+```protobuf
+/**
+ * An addition of a new named state.
+ */
+message NewStateChange {
+    /**
+     * The type (e.g. Singleton, Virtual Map, Queue) of state to add.
+     */
+    NewStateType state_type = 1;    
+}
+
+/**
+ * An enumeration of the types of named states.
+ */
+enum NewStateType {
+    SINGLETON = 0;
+    VIRTUAL_MAP = 1;
+    QUEUE = 2;
+}
+```
+
+```protobuf
+/**
+ * A removal of a named state.
+ */
+message RemovedStateChange {
+}
+```
+
+```protobuf
+/**
+ * An update to a `Singleton` state.
+ */
+message SingletonUpdateChange {
+		oneof new_value {
+				/**
+		     * A change to the block info singleton.
+		     */
+		    proto.BlockInfo block_info_value = 1;
+
+		    /**
+		     * A change to the congestion level starts singleton.
+		     */
+		    proto.CongestionLevelStarts congestion_level_starts_value = 2;
+
+		    /**
+		     * A change to the Entity Identifier singleton.
+		     */
+		    google.protobuf.UInt64Value entity_number_value = 3;
+
+		    /**
+		     * A change to the exchange rates singleton.
+		     */
+		    proto.ExchangeRateSet exchange_rate_set_value = 4;
+
+		    /**
+		     * A change to the network staking rewards singleton.
+		     */
+		    proto.NetworkStakingRewards network_staking_rewards_value = 5;
+
+		    /**
+		     * A change to a raw byte array singleton.
+		     */
+		    google.protobuf.BytesValue bytes_value = 6;
+
+		    /**
+		     * A change to a raw string singleton.
+		     */
+		    google.protobuf.StringValue string_value = 7;
+
+		    /**
+		     * A change to the running hashes singleton.
+		     */
+		    proto.RunningHashes running_hashes_value = 8;
+
+				/**
+         * A change to the throttle usage snapshots singleton.
+         * <p>
+         * Throttle usage snapshots SHALL be updated for _every transaction_
+         * to reflect the amount used for each tps throttle and
+         * for the gas throttle.
+         */
+		    proto.ThrottleUsageSnapshots throttle_usage_snapshots_value = 9;
+
+				/**
+         * A change to a raw `Timestamp` singleton.<br/>
+         * An example of a raw `Timestamp` singleton is the
+         * "network freeze time" singleton state, which, if set, stores
+         * the time for the next scheduled freeze.
+         */
+        proto.Timestamp timestamp_value = 10;
+
+        /**
+         * A change to the block stream status singleton.
+         */
+        com.hedera.hapi.node.state.blockstream.BlockStreamInfo block_stream_info_value = 11;
+
+        /**
+         * A change to the platform state singleton.
+         */
+        com.hedera.hapi.platform.state.PlatformState platform_state_value = 12;
+
+        /**
+         * A change to the roster state singleton.
+         */
+        com.hedera.hapi.node.state.roster.RosterState roster_state_value = 13;
+		}
+}
```

Comment:
A swarm of tab characters seem to have snuck in to this block.
```suggestion
    oneof new_value {
        /**
         * A change to the block info singleton.
         */
        proto.BlockInfo block_info_value = 1;

        /**
         * A change to the congestion level starts singleton.
         */
        proto.CongestionLevelStarts congestion_level_starts_value = 2;

        /**
         * A change to the Entity Identifier singleton.
         */
        google.protobuf.UInt64Value entity_number_value = 3;

        /**
         * A change to the exchange rates singleton.
         */
        proto.ExchangeRateSet exchange_rate_set_value = 4;

        /**
         * A change to the network staking rewards singleton.
         */
        proto.NetworkStakingRewards network_staking_rewards_value = 5;

        /**
         * A change to a raw byte array singleton.
         */
        google.protobuf.BytesValue bytes_value = 6;

        /**
         * A change to a raw string singleton.
         */
        google.protobuf.StringValue string_value = 7;

        /**
         * A change to the running hashes singleton.
         */
        proto.RunningHashes running_hashes_value = 8;

        /**
         * A change to the throttle usage snapshots singleton.
         * <p>
         * Throttle usage snapshots SHALL be updated for _every transaction_
         * to reflect the amount used for each tps throttle and
         * for the gas throttle.
         */
        proto.ThrottleUsageSnapshots throttle_usage_snapshots_value = 9;

        /**
         * A change to a raw `Timestamp` singleton.<br/>
         * An example of a raw `Timestamp` singleton is the
         * "network freeze time" singleton state, which, if set, stores
         * the time for the next scheduled freeze.
         */
        proto.Timestamp timestamp_value = 10;

        /**
         * A change to the block stream status singleton.
         */
        com.hedera.hapi.node.state.blockstream.BlockStreamInfo block_stream_info_value = 11;

        /**
         * A change to the platform state singleton.
         */
        com.hedera.hapi.platform.state.PlatformState platform_state_value = 12;

        /**
         * A change to the roster state singleton.
         */
        com.hedera.hapi.node.state.roster.RosterState roster_state_value = 13;
    }
}
```

Reply:
Updated